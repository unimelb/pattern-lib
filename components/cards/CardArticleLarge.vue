<template>
  <div class="card-article-large">
    <div class="card-article-large__inner">
      <div class="card-article-large__content">
        <h3 class="heading-md">
          {{ title }}
        </h3>
        <p class="card-article-large__category">
          {{ category }}
        </p>
        <p
          v-if="description"
          class="content-max-width">
          {{ description }}
        </p>
        <div>
          <ButtonIcon
            v-if="href"
            :href="href"
            class="btn--text"
            :aria-label="`Read more about ${title}`">
            Read more
          </ButtonIcon>
        </div>
      </div>
      <div class="card-article-large__img">
        <client-only>
          <progressive-img
            :src="img.src"
            :alt="img.alt || ''"
            :aspect-ratio="0.675" />
        </client-only>
      </div>
    </div>
  </div>
</template>

<script>
import ButtonIcon from 'components/buttons/ButtonIcon.vue';

export default {
  components: {
    ButtonIcon,
  },
  props: {
    title: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      required: false,
      default: '',
    },
    category: {
      type: String,
      required: false,
      default: '',
    },
    img: {
      type: Object,
      required: false,
      default() {
        return null;
      },
    },
    href: {
      type: String,
      required: false,
      default: '',
    },
  },
};
</script>
