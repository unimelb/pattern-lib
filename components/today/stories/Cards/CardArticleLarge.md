## Card Article Large

The component `CardArticleLarge` acceps the following props:

| props    	   | type    	|
|------------- |---------	|
| href   	     | String  	|
| title    	   | String  	|
| category     | String   |
| description  | String   |
| img          | Object   |

#### HTML Template

```html
  <card-article-large
    href="https://www.unimelb.edu.au/"
    title="Test title"
    category="Test category"
    description="Nisi labore mollit occaecat nisi et irure esse mollit veniam in."
    :img="{ src: 'https://picsum.photos/seed/1/540/360', alt: 'Image for example purposes.' }">
    </card-article-large>
```