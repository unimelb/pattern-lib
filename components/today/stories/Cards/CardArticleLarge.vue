<template>
  <CardArticleLarge
    href="https://www.unimelb.edu.au/"
    category="Latest research"
    title="How Indigenous knowledge can save our environment"
    description="Wiradjuri man and geographer Associate Professor Michael-Shawn Fletcher explores the guiding principles of landscape management to create resources or resource availability in a landscape and to provide a secure place to live."
    :img="{ src: 'https://picsum.photos/seed/1/540/360', alt: 'Image for example purposes.' }" />
</template>

<script>
import CardArticleLarge from 'components/cards/CardArticleLarge.vue';
import CardArticleLargeMd from './CardArticleLarge.md';

export default {
  components: {
    CardArticleLarge,
  },
  readme: {
    custom: CardArticleLargeMd,
    source: true,
    html: true,
  },
};
</script>
