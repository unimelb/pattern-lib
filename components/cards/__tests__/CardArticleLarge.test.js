import {
  mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardArticleLarge from '../CardArticleLarge.vue';

expect.extend(toHaveNoViolations);

describe('CardArticleLarge', () => {
  it('should match snapshot', () => {
    const result = mount(CardArticleLarge, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
        category: 'mock category',
        img: "{ src: 'https://picsum.photos/seed/1/540/360', alt: 'test alt' }",
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(CardArticleLarge, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
        category: 'mock category',
        img: "{ src: 'https://picsum.photos/seed/1/540/360', alt: 'test alt' }",
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
