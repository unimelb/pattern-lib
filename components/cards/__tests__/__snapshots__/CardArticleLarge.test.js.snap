// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardArticleLarge should match snapshot 1`] = `
<div
  class="card-article-large"
>
  <div
    class="card-article-large__inner"
  >
    <div
      class="card-article-large__content"
    >
      <h3
        class="heading-md"
      >
        
        mock title
      
      </h3>
       
      <p
        class="card-article-large__category"
      >
        
        mock category
      
      </p>
       
      <!---->
       
      <div>
        <a
          aria-label="Read more about mock title"
          class="btn--text btn"
          href="www.unimelb.edu.au"
        >
          <span
            class="push-icon"
          >
            
          Read more
         
            <svg
              aria-hidden="true"
              aria-label="chevron-right"
              class="push-icon__icon"
              focusable="false"
              height="15px"
              role="img"
              width="15px"
            >
              <use
                xlink:href="#icon-chevron-right"
              />
               
              <title>
                Chevron-right
              </title>
               
            </svg>
             
            <!---->
          </span>
           
          <!---->
        </a>
      </div>
    </div>
     
    <div
      class="card-article-large__img"
    >
      <client-only>
        <progressive-img
          alt=""
          aspect-ratio="0.675"
        />
      </client-only>
    </div>
  </div>
</div>
`;
